store/src/java/com/zimbra/cs/mailbox/MailItem.java

```diff
@@ -2703,6 +2712,10 @@ MailItem copy(Folder folder, int copyId, String copyUuid, MailItem parent) throw
      *    <li><tt>service.PERM_DENIED</tt> - if you don't have sufficient
      *        permissions</ul> */
     MailItem icopy(Folder target, int copyId, String copyUuid) throws IOException, ServiceException {
+        return icopy(target, copyId, copyUuid, 0);
+    }
+
+    MailItem icopy(Folder target, int copyId, String copyUuid, int trashFolderId) throws IOException, ServiceException {
```

Comment:
same

Reply:
replaced the trashFolderId with the boolean values
done for all overloaded method